### ISSUE TITLE
Hand picked product ordering / sort

### ISSUE THREAD

**Issue by WiredWonder** [woocommerce/woocommerce#42663]
## Is your feature request related to a problem? Please describe.
When choosing items for the Hand Picked Product block, they are sorted by the default Woocommerce sort order options (name, newness, etc).

## Describe the solution you'd like
It would be amazing if the Hand Picked Product block allowed the author to custom sort the order order of the products in the widget, ideally through something like drag and drop of the items in the selector.

## Describe alternatives you've considered
na

## Additional context
na

**Comment by Aljullu** [comment]
Thanks for opening this issue @galapogos01. While the Hand-picked Products block doesn't allow ordering products right inside the block, you can tweak the order products are shown as follows:

1. In the Hand-pick Products block, select the _Order products by: Menu order_ option.

<img src="https://user-images.githubusercontent.com/3616980/133965987-c57c4050-e756-4d30-a685-5e611c8f5567.png" width="293" alt="Order products by: Menu order option" />

2. Now, in wp-admin, go to _Products_.
3. There, switch to the _Sorting_ tab:

<img src="https://user-images.githubusercontent.com/3616980/133966061-ab5833e5-e462-430b-be06-a6bffa2780c8.png" width="352" alt="Sorting products" />

4. You will be able to reorder the products there using drag and drop. That order will be honored in the Hand-picked Products block you just added, as well as other parts of WooCommerce like the Shop page.

You can read more on how to do it from this page:

https://docs.woocommerce.com/document/managing-products/#sorting-products

Hope it helps! I will proceed closing this issue, but feel free to comment again if needed.

**Comment by WiredWonder** [comment]
@Aljullu can you please re-open the ticket? The proposed solution changes the global product sort order everywhere, and is not a viable solution for an actual store with 4500 products.

Thanks,
Jason

**Comment by Aljullu** [comment]
Fair enough. I will keep this issue open, but to make expectations clear, we don't have plans to implement something like this directly in the block in the short-term.

**Comment by onthisearth** [comment]
@Aljullu Would love to see this implemented, too.

**Comment by WiredWonder** [comment]
Please don't close this, I'm not the only person asking for it :)

**Comment by ijgeoffrey** [comment]
I definitely feel like this should be a feature. Why hand-pick products without hand-picking order? As it is I'm gonna have to try to find a workaround, and for reasons mentioned above, menu order is not a viable solution.

**Comment by katiebethbrown** [comment]
Adding a +1 as we ran into this on a Team 51 partner site. We were able to work around it with the menu order, but it would be nice to be able to set the order explicitly in the block without impacting the other areas of the site, similar to changing the order of images in a gallery.

**Comment by tommusrhodus** [comment]
@Aljullu I have added a PR that adds basic sorting abilities to this block woocommerce/woocommerce-blocks#6666 to assist with the partner site mentioned by @katiebethbrown

**Comment by tommusrhodus** [comment]
Whilst my PR above would not work with the direction that WooCommerce Blocks is heading in, I have added my Proof of Concept to a seperate repo that can be used by anyone to allow sorting this block as a stop-gap until future updates to the core plugin:

https://github.com/Automattic/handpicked-products-order

Note: This stop-gap is exactly that, it just fills a need for now, this plugin will become obsolete in future WooCommerce Blocks updates.

**Comment by jamesckemp** [comment]
Thanks for submitting an issue, and sorry it took so long to review it! The Hand-Picked Products collection in Product Collection supports this ordering without changing the global catalog order; if you're using the legacy Hand-picked Products block, switch to that collection to use it. I'll close this for now, but if you think we've missed something, please follow up here.

**Comment by WiredWonder** [comment]
@jamesckemp can you explain why there are two blocks called Hand-Picked Products, and how to tell which one is which?

**Comment by kmanijak** [comment]
> can you explain why there are two blocks called Hand-Picked Products, and how to tell which one is which?

Hey @WiredWonder, that's rather unexpected 🤔 Do you mind sharing what you see in the inserter when trying to insert "Hand-Picked products"?

<img width="324" height="134" alt="Image" src="https://github.com/user-attachments/assets/2693fa26-d119-407b-984d-524d2a20e9cb" />

Ideally, if you also could insert both blocks, switch to Code Editor (Three-dots menu > Code Editor) and share the content so we can see what block exactly shows up for you.

<img width="332" height="437" alt="Image" src="https://github.com/user-attachments/assets/7bf8683c-9771-4fba-b66a-72408f1b1727" />

Thanks! 🙌

**Comment by WiredWonder** [comment]
@kmanijak I am responding to @jamesckemp who said:

> The Hand-Picked Products collection in Product Collection supports this ordering without changing the global catalog order; if you're using the legacy Hand-picked Products block, switch to that collection to use it. 

Why are there two in the first instance? 
As a user, how am I supposed to tell them apart, or that a new one even was added to WooCommerce?
If one has now replaced the other, why didn't you migrate content to use the newer one?
edit: further, having just played with this new block, it behaves a little strangely. For example it has separate elements that can be selected but it's not entirely clear which can be edited without breaking the block. Most importantly though, it is not obvious how to sort the products -- which is why I raised the Issue.

**Comment by kmanijak** [comment]
I see, I misunderstood your initial question! 😅 

> Why are there two in the first instance?

Good point, earlier there was couple of product grid blocks including Handpicked Products that [we deprecated](https://developer.woocommerce.com/2024/11/19/product-collection-block-the-new-default/) in favor of Product Collection block. That means the "older" blocks were hidden from inserter but kept functional for those that kept using them.

> As a user, how am I supposed to tell them apart

"Older" product grid blocks had limited customisation options with no way of changing layout of your product grid. The new one, based on Product Collection, provides more customisation settings and it's built with smaller blocks allowing to create advanced and unusual product layout.

Legacy Handpicked Products | Current Hand-picked Products
-- | --
<img width="1032" height="716" alt="Image" src="https://github.com/user-attachments/assets/c481da78-a5c4-4912-b32e-ceda2b46f178" /> | <img width="1211" height="1093" alt="Image" src="https://github.com/user-attachments/assets/3240af6b-3cae-47b8-bd46-8c7a7fb32eef" />

> or that a new one even was added to WooCommerce?

We communicated it via blog posts and in GitHub in advance, for example:
- [Announcing the Product Collection Block](https://developer.woocommerce.com/2024/02/22/announcing-the-product-collection-block/)
- [Product Collection block—The new default](https://developer.woocommerce.com/2024/11/19/product-collection-block-the-new-default/)
- [Product Collection block: New JS events and soft deprecation of Product Grid blocks](https://developer.woocommerce.com/2024/11/11/product-collection-block-new-js-events-and-soft-deprecation-of-product-grid-blocks/)
- [Making Product Collection the main block for product lists and collections](https://github.com/woocommerce/woocommerce/discussions/44776)

At the same time, the editor transition was meant to be seamless: we hidden "older" blocks and exposed "newer" ones at the same time.

> If one has now replaced the other, why didn't you migrate content to use the newer one?

There's few aspects, that made it risky:
- These blocks are not exactly 1:1 equivalents and hence migration could cause some visual or extensibility differences,
- There's no official Gutenberg way of migrating one block to another (see [Deprecation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/)) and hence that would require custom mechanism that goes through users pages, posts, templates, patterns and whatnot, find blocks and replace them. There's so many ways in which this could go wrong.

So to prioritise backwards compatibility and stability of stores we decided not to pursue this path.

> For example it has separate elements that can be selected but it's not entirely clear which can be edited without breaking the block.

Any of the product elements: image, price, etc. can be edited, moved, added, removed and Product Collection will work. You can also add other blocks to it, e.g. SKU or Product Brands/Categories. The important bit is Product Template - that's needed to be there.

> Most importantly though, it is not obvious how to sort the products -- which is why I raised the Issue.

Fair feedback. Products are ordered in insertion order which is limiting and annoying, I must agree. I agree this issue should be re-opened and the ordering should be revisited.

Drag and drop sounds like an obvious choice _but_ it leads to other problems (documenting for engineer that will pick it up):
- Product Collection is paginated and we only display one page in editor - sorting requires to show all of the products in grid.
- With interactive inner blocks, drag and drop would compete for mouse interactions - we need to take into account.
- I'd consider a dedicated "sorting" mode for Handpicked Products, when clicked we'd replace Product Collection content with "sorting modal", similar to collection initial state but dedicated to sorting:

<img width="680" height="653" alt="Image" src="https://github.com/user-attachments/assets/7dd03769-c196-4f57-8f89-e4bec63d357a" />

 or at least drag and drop in `FormTokenField` (upstream contribution):

<img width="289" height="349" alt="Image" src="https://github.com/user-attachments/assets/52c8359a-0597-43db-b435-bee8fb5f439e" />

But anyway, I'm reopening the issue as I consider it valid 🙌

**Comment by WiredWonder** [comment]
Thank you @kmanijak for taking the time to provide a detailed reply to my message, I really appreciate it!

It all makes sense, but I will give one piece of feedback -- from the perspective of a busy business owner/store operator/content author, those developer blog entries do not cut it as notification. If not possible via an admin notice or other mechanism at upgrade time, it would be great if the block UI somehow alerted the author that the block in their page was deprecated and suggested the replacement.

Given the age of Wordpress/Woo there are many old sites that predate huge new features being added; things like Brand, GTIN, HPOS and others come to mind; the Woo ecosystem could possibly keep a lot more stores from migrating to other platforms by gently nudging us pesky users along the adoption path. 

My 2c delivered, I will watch this Issue to see if we get a fix for Ordering.

Thanks, Jason
